test.tf

```diff
@@ -1,171 +1,276 @@
-  degraded_response_time = 15000
-  max_response_time      = 30000
-
-  locations = [
-    "us-west-1",
-    "ap-northeast-1",
-    "ap-south-1",
-  ]
-
   alert_settings {
     escalation_type = "RUN_BASED"
-
+    reminders {
+      amount = 0
+      interval = 5
+    }
     run_based_escalation {
       failed_run_threshold = 1
     }
-
-    time_based_escalation {
-      minutes_failing_threshold = 5
-    }
-
     ssl_certificates {
-      enabled         = true
       alert_threshold = 30
+      enabled = true
     }
-
-    reminders {
-      amount = 1
+    time_based_escalation {
+      minutes_failing_threshold = 5
     }
   }
+}
+
+#----------------------------EXAMPLE----------------------------#
+#-- a POST API Check with json body
+
+resource "checkly_check" "canonical-api-check-3" {
+  name = "API check 3"
+  type = "API"
+  activated = true
+  double_check = true
+  frequency = 720
+  locations = [
+    "eu-central-1",
+    "us-east-2",
+  ]
+  max_response_time = 18000
+  muted = true
+  environment_variables = null
 
   request {
+    method = "POST"
+    url = "https://jsonplaceholder.typicode.com/posts"
     follow_redirects = true
-    url              = "http://example.com/"
-
-    query_parameters = {
-      search = "foo"
-    }
 
     headers = {
-      X-Bogus = "bogus"
+        Content-type = "application/json; charset=UTF-8"
     }
 
+    body = "{\"message\":\"hello checkly\",\"messageId\":1}"
+    body_type = "JSON"
+
     assertion {
-      source     = "JSON_BODY"
-      property   = "code"
-      comparison = "HAS_VALUE"
-      target     = "authentication.failed"
+      comparison = "EQUALS"
+      property = ""
+      source = "STATUS_CODE"
+      target = "201"
     }
 
     assertion {
-      source     = "STATUS_CODE"
-      property   = ""
       comparison = "EQUALS"
-      target     = "401"
+      source = "JSON_BODY"
+      property = "$.message"
+      target = "hello checkly"
     }
 
-    basic_auth {
-      username = ""
-      password = ""
+    assertion {
+      comparison = "EQUALS"
+      source = "JSON_BODY"
+      property = "$.messageId"
+      target = 1
     }
+
   }
-  group_id    = checkly_check_group.test-group1.id
-  group_order = 2
+
+  use_global_alert_settings = true
+
 }
 
-resource "checkly_check_group" "test-group1" {
-  name      = "My test group 1"
+################################# BROWSER CHECKS ################################# 
+
+#----------------------------EXAMPLE----------------------------#
+#-- a Browser Check which runs E2E test
+
+resource "checkly_check" "browser-check-1" {
+  name                      = "A simple browser check"
+  type                      = "BROWSER"
+  activated                 = true
+  should_fail               = false
+  frequency                 = 10
+  double_check              = true
+  ssl_check                 = true
+  use_global_alert_settings = true
+  locations = [
+    "us-west-1"
+  ]
+
+  script = <<EOT
```

Comment:
Because it's common for browser check scripts to be more complicated than this, and because users often want to use the same script to check multiple endpoints, it might be a neat idea to show how to source the script from a file:

```terraform
data "local_file" "browser-script" {
  filename = "${path.module}/browser-script.js"
}

resource "checkly_check" "browser-check-1" {
  ...
  script = data.local_file.browser-script.content
}
```

Reply:
That is neat. Will add it.